packages/components/upload/src/upload-content.vue

```diff
@@ -92,9 +92,15 @@ const upload = async (rawFile: UploadRawFile) => {
   let beforeData: UploadContentProps['data'] = {}
 
   try {
+    // origin data: Handle data changes after synchronization tasks are executed
+    const originData = props.data
     const beforeUploadPromise = props.beforeUpload(rawFile)
     beforeData = isObject(props.data) ? cloneDeep(props.data) : props.data
     hookResult = await beforeUploadPromise
+    beforeData =
+      isObject(props.data) && isEqual(originData, beforeData)
+        ? cloneDeep(props.data)
+        : beforeData
```

Comment:
It seems that it can be adjusted as follows:
```ts
if (isObject(props.data) && isEqual(originData, beforeData)) {
  beforeData = cloneDeep(props.data)
}
```